spec/03_result_for_api_spec.rb

```diff
@@ -4,7 +4,24 @@
 
 RSpec.describe 'Result for API' do
   def result_for_api(status, api)
-    # TODO
+    case [status, api]
+    in 200, { success: false }
+      Result::Failure.new([])
+    in 200, {}
+      Result::Success.new(nil)
+    in 200, { payload: { **payload } }
```

Comment:
Je pense que tu peux simplement utiliser

```suggestion
    in 200, { payload: }
```

Reply:
effectivement, pas mal comme ça !